layer/mounted_layer.go

```diff
@@ -50,11 +53,68 @@ func (ml *mountedLayer) Parent() Layer {
 	return nil
 }
 
+func (ml *mountedLayer) incActiveMount() {
+	ml.activeMountL.Lock()
+	ml.activeMount++
+	ml.activeMountL.Unlock()
+}
+
+func (ml *mountedLayer) decActiveMount() {
+	ml.activeMountL.Lock()
+	ml.activeMount--
+	ml.activeMountL.Unlock()
+}
+
 func (ml *mountedLayer) Mount(mountLabel string) (string, error) {
-	return ml.layerStore.driver.Get(ml.mountID, mountLabel)
+	var path string
+	var err error
+
+	ml.activeMountL.Lock()
+	activeMount := ml.activeMount
+	ml.activeMountL.Unlock()
```

Comment:
This lock needs to be held for the whole(majority) of the method. Otherwise, parallel calls may still call underlying methods in wrong order.